### PR TITLE
raw3 3 canopen

### PR DESCRIPTION
**Pull request by fmessmer**
this PR migrates `raw3-3` base to `ros_canopen`
see #422 

thanks @ipa-jba and @ipa-mdl for doing this!

### CODE REVIEW

**Comment by fmessmer** [comment]
@destogl @andreeatulbure 
this might be of interest for you as we made some consistency changes within the (raw3-common) raw3_base_driver.yaml file which is also used on your `raw3-5`
Could you quickly test this branch and let us know whether it works for you? Thanks

**Comment by fmessmer** [comment]
we need to make the `steer_ctrl` parameters of the `cob_omni_drive_controller/WheelController` robot-specifc:
for raw3-3 they previously have been set to
```
    steer_ctrl: # Parameters for tuning Impedance-Controller
      spring: 15.0 #10.0
      damp: 2.75 #2.5
      virt_mass: 0.1 #0.1
      d_phi_max: 12.0 #100.0
      dd_phi_max: 40.0 #100.0
```


but in `raw3_base_controller.yaml` they are now:
```
     steer_ctrl: # Parameters for tuning Impedance-Controller
       spring: 9.5
       damp: 2.25
       virt_mass: 0.1 
       d_phi_max: 12.0 #100.0
       dd_phi_max: 40.0 #100.0
```

**Comment by mathias-luedtke** [comment]
Three options:
1. try to make them the same
2. Split config (upload additional params in `raw3_base.launch`)
3. separate config (with [controller_yaml](https://github.com/ipa320/cob_robots/blob/indigo_dev/cob_bringup/components/raw3_base.launch#L10) arg)

**Comment by fmessmer** [comment]
@ipa-jba will post a few comments about the steps to-be-done to setup the pcan driver in #422

**Comment by jabbenseth** [comment]
see https://github.com/ipa320/cob_robots/issues/422#issuecomment-297630074

**Comment by fmessmer** [comment]
@ipa-jba 
I added another commit that should handle the control parameter missmatch between raw3-3 and raw3-5
please retest

@ipa-mdl 
I opted for this approach as I did not want to add `base_controller.yaml` files back to all the other robots which seem to work fine with the default values...
What do you think?

**Comment by fmessmer** [comment]
The parameter handling did not fully solve the oscillation....we'll do an evaluation analogue to https://github.com/ipa320/cob_control/issues/143

**Comment by jabbenseth** [comment]
Right now the base starts to oscillate on the drive wheels. See the attached screenshot (shown is a rotation, where oscillation is more visible, but not exclusive)
screenshot from 2017-04-27 14 43 57

This is only reproducible if the base is on the floor (not hanging)

**Comment by fmessmer** [comment]
according to @ipa-tif this is due to hitting an resonance frequency with the drive wheels working against the springs. a similar behavior has been observed on cob4 with loosely tuned elmo controllers

thus, it is **not** an issue of `ros_canopen` and/or configuration but an issue of the internal elmo controller configuration.

next step would be to investigate to what extend the elmo controllers can be tuned
the old elmo configuration laptop is put on `raw3-3` - and either @ipa-tif or @ipa-bdw are able to check this

**Comment by fmessmer** [comment]
Another issue was with emergency stop being triggered which required modifications in `ros_canopen` see https://github.com/ros-industrial/ros_canopen/pull/234

**Comment by fmessmer** [comment]
@ipa-mdl ready for final review
@ipa-jba @ipa-mig ready for final test 
(note: this branch is rebased...use `git checkout -B raw3-3_canopen origin-fxm/raw3-3_canopen`)

I'll merge once I get your approvals

File: cob_bringup/drivers/canopen_402.launch
Comment (mathias-luedtke) [inline cob_bringup/drivers/canopen_402.launch]: this does not belong here

File: cob_bringup/components/ros_control_base.launch
Comment (mathias-luedtke) [inline cob_bringup/components/ros_control_base.launch:17]: This is the right place, I don't know why it was below beforehand.
It might break other robots with outdated configs..

File: cob_bringup/robots/raw3-3.xml
Comment (mathias-luedtke) [inline cob_bringup/robots/raw3-3.xml:54]: `can0` is the default

File: cob_bringup/robots/raw3-5.xml
Comment (mathias-luedtke) [inline cob_bringup/robots/raw3-5.xml]: This looks wrong, controller config should go into `base` namespace

File: cob_hardware_config/robots/raw3-5/config/base_controller.yaml
Comment (mathias-luedtke) [inline cob_hardware_config/robots/raw3-5/config/base_controller.yaml:6]: the last three could be omited, since the are same as in the default config

**Review by mathias-luedtke — CHANGES_REQUESTED** [review]

File: cob_hardware_config/robots/raw3-5/config/base_controller.yaml
Comment (fmessmer) [reply cob_hardware_config/robots/raw3-5/config/base_controller.yaml:6]: let's keep the parameters together...

**Comment by fmessmer** [comment]
I considered @ipa-mdl's comments and tested this branch (including https://github.com/ros-industrial/ros_canopen/pull/234) onc `cob4-9`...works as expected

I will do a final test with @ipa-jba on `raw3-3` to clarify https://github.com/ipa320/cob_robots/pull/661#discussion_r115229481

**Comment by mathias-luedtke** [comment]
merged the reset feature.

File: cob_bringup/robots/raw3-3.xml
Comment (fmessmer) [reply cob_bringup/robots/raw3-3.xml:54]: but the argument is required...

**Pull request merged**
